### PR TITLE
Add support for Universal Linux Client

### PR DESCRIPTION
**Pull request by arogge**
This PR introduces a new build-type "Universal Client" that can be used to compile a filedaemon with only a few dependencies.
Currently you can build and package for Debian.
The resulting packages are currently untested and packaging still needs to be changed so a "real" client will automatically be an upgrade to the universal client.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
- [x] Merge branch PR-1019 into master of CD repository before merging this one!

##### General

- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [ ] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems
- [x] `git status` should not report modifications in the source tree after building and testing

### CODE REVIEW

File: core/src/stored/bscan.cc [hunk 1]
```diff
@@ -1090,28 +1090,14 @@ static bool CreateFileAttributesRecord(BareosDb* db,
...
-    dt.julian_day_number = vl->write_date;
-    dt.julian_day_fraction = vl->write_time;
-    TmDecode(&dt, &tm);
-    mr->FirstWritten = mktime(&tm);
-    dt.julian_day_number = vl->label_date;
-    dt.julian_day_fraction = vl->label_time;
-    TmDecode(&dt, &tm);
-    mr->LabelDate = mktime(&tm);
-  }
+  ASSERT(vl->VerNum >= 11);
```
Comment (joergsteffens) [inline core/src/stored/bscan.cc:1097]: A comment about this ASSERT might be useful.

File: core/cmake/BareosGenerateDebianInfo.cmake [hunk 2]
```diff
@@ -17,98 +17,161 @@
...
-         DEBIAN_CONTROL_FILEDAEMON_PYTHON3_PLUGIN
-    )
-    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-storage-python3-plugin
-         DEBIAN_CONTROL_STORAGE_PYTHON3_PLUGIN
-    )
-    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-director-python3-plugin
-         DEBIAN_CONTROL_DIRECTOR_PYTHON3_PLUGIN
-    )
-  endif()
+  if(GENERATE_DEBIAN_CONTROL)
```
Comment (joergsteffens) [inline core/cmake/BareosGenerateDebianInfo.cmake:64]: Is this change of logic intentional ?

File: core/CMakeLists.txt [hunk 3]
```diff
@@ -632,14 +651,7 @@ set(have_crypto "${HAVE_CRYPTO}")
...
 
-set(HAVE_CAPABILITY
-    "${HAVE_CAPABILITY_H}"
-    CACHE BOOL "compile with capability support"
-)
-if("${HAVE_CAPABILITY}")
-  set(CAPABILITY_LIBS cap)
-  message(STATUS "CAPABILITY set")
-endif()
+option(ENABLE_CAPABILITY "Enable capability support" "${HAVE_CAPABILITY_H}")
```
Comment (joergsteffens) [inline core/CMakeLists.txt:654]:
```suggestion
option(ENABLE_CAPABILITY "Enable capability-based security support" "${HAVE_CAPABILITY_H}")
```

File: core/src/stored/bscan.cc (hunk 1 above)
Comment (arogge) [reply core/src/stored/bscan.cc:1098]: +1

File: core/CMakeLists.txt (hunk 3 above)
Comment (arogge) [reply core/CMakeLists.txt:654]: Right now this enables/disables support for the `-k` option on the FD.
Maybe we should call it "Enable keep readall capabilities in filedaemon", as that is what this actually switches.

Comment (joergsteffens) [reply core/CMakeLists.txt:654]: If it is only used in the FD: yes, that would be helpful.

File: core/cmake/BareosGenerateDebianInfo.cmake (hunk 2 above)
Comment (arogge) [reply core/cmake/BareosGenerateDebianInfo.cmake]: I completely changed the creation of `debian/control`.
I think the new variant is easier to understand - what do you think?

Comment (joergsteffens) [reply core/cmake/BareosGenerateDebianInfo.cmake]: Yes.

**Review by pstorz — APPROVED** [review]
All changes look good and the tests of the ULC also all are successful.

**Review by joergsteffens — APPROVED** [review]
Approved as discussed. Documentation wil be part of a separate PR.

**Review by pstorz — APPROVED** [review]

**Pull request merged**